Assets/__Scripts/Map/Events/EventAppearanceSO.cs

```diff
@@ -132,27 +147,34 @@ public class EventAppearanceSO : ScriptableObject
                 e.ChangeBaseColor(color, false);
                 break;
             case MapEvent.LIGHT_VALUE_BLUE_FLASH:
-                e.UpdateOffset(e.UsePyramidModel ? pyramidFlashShaderOffset : cubeFlashShaderOffset, false);
+                e.UpdateOffset(flashShaderOffset[e.EventModel], false);
                 break;
             case MapEvent.LIGHT_VALUE_BLUE_FADE:
-                e.UpdateOffset(e.UsePyramidModel ? pyramidFadeShaderOffset : cubeFadeShaderOffset, false);
+                e.UpdateOffset(fadeShaderOffset[e.EventModel], false);
                 break;
             case MapEvent.LIGHT_VALUE_RED_ON:
                 e.UpdateOffset(Vector3.zero, false);
                 e.ChangeBaseColor(color, false);
                 break;
             case MapEvent.LIGHT_VALUE_RED_FLASH:
-                e.UpdateOffset(e.UsePyramidModel ? pyramidFlashShaderOffset : cubeFlashShaderOffset, false);
+                e.UpdateOffset(flashShaderOffset[e.EventModel], false);
                 break;
             case MapEvent.LIGHT_VALUE_RED_FADE:
-                e.UpdateOffset(e.UsePyramidModel ? pyramidFadeShaderOffset : cubeFadeShaderOffset, false);
+                e.UpdateOffset(fadeShaderOffset[e.EventModel], false);
                 break;
         }
 
-        e.ChangeFadeSize(e.UsePyramidModel ? pyramidDefaultFadeSize : cubeDefaultFadeSize, false);
+        e.ChangeFadeSize(defaultFadeSize[e.EventModel], false);
 
         if (Settings.Instance.VisualizeChromaGradients) e.UpdateGradientRendering();
 
         e.UpdateMaterials();
     }
 }
+
+public enum EventModelType
```

Comment:
Unless I'm missing something this is the most wasted enum. You use int everywhere and only cast from this enum once.
(You could be using it in a scene somewhere I haven't checked)
But you don't even use it in UpdateOldSettings???